client/dashboard/sites/settings-page-header/index.tsx

```diff
@@ -15,19 +15,32 @@ export default function SettingsPageHeader( props: SettingsPageHeaderProps ) {
 	const { siteSlug } = siteRoute.useParams();
 	const { backPath, backLabel, ...pageHeaderProps } = props;
 	const router = useRouter();
+	const canGoBack = useCanGoBack();
 
 	const defaultBackPath = `/sites/${ siteSlug }/settings`;
-	const defaultBackLabel = __( 'Settings' );
+	const defaultBackLabel = __( 'Back' );
+	const fallbackBackLabel = __( 'Settings' );
+
+	const getLabel = () => {
+		if ( backLabel ) {
+			return backLabel;
+		}
+		return canGoBack ? defaultBackLabel : fallbackBackLabel;
```

Comment:
The problem of this way is https://github.com/Automattic/wp-calypso/pull/104882#discussion_r2225102072. We want to keep showing the "Settings" when user navigates from the "Settings" to its child pages.

However, when trying to resolve this issue, https://github.com/Automattic/wp-calypso/pull/104882 became more complicated than I expected, so I ended up closing it at that time. 🥲